### ISSUE TITLE
lost access rights to put after multiple users have root directory in the directory server

### ISSUE THREAD

**Issue by manikawnth** [upspin/upspin#587]
I've created a user manikawnth@gmail.com via `upspin-ui`

I created an upspin server `upspin.manikawnth.com`  on aws ec2 with the storage backend as dropbox.
Things are working fine, even without Access file (for self), I was able to put resources and get other publicly available files.

Note: My client machine is windows 7 and using upspin-ui for file uploading and exploring

Now, I created a second user manikawnth@yahoo.co.in (from windows 7 `upspin` command line) and I updated the `Writers` file on upspin server.  I then restarted the systemd upspinserver.service and I pointed the config file in my `$HOME\upspin\config` to the corresponding 2nd user and everything's working fine.

Then, I pointed the `$HOME\upspin\config` to my first user (gmail user) and then started the `upspin-ui` . Now *I lost all the put access* . Its throwing the below error message:

>dir/remote("upspin.manikawnth.com:443").Put: permission denied: dir/server.Put: manikawnth@gmail.com/README

Please help out if I'm doing something wrong.

**Comment by adg** [comment]
Did you include manikawnth@gmail.com in the server writers file? No typos?

Did you put an Access file at manikawnth@gmail.com/Access ? If so did you
include yourself?

Check that upspin-ui is really running as who you think it is. What does
the name in the top left say?

Finally, what is your output from the command "upspin config"?

Andrew

**Comment by manikawnth** [comment]
Hey @adg ,

I figured out two things.
Writing anything in the `Writers` file is not really working. I had to use `upspin setupwriters` command to get it working.
Also, I was under the impression, irrespective of what's in the Access file, the root owner will always have the full access to his root hierarchy. But that's wrong. I had to explicitly put my username in the `Access` file.
With these two things (setupwriters command, modifying access file), things are working now

**Comment by kushtrimjunuzi** [comment]
Hi @adg ,
Same thing happens to me by accident I did not put root username into Access file and now I'm not able to modify that file, this root user is in Writers file but that does not work. I have tried steps from @manikawnth above but still same thing, how can you recover from this issue? Btw this Access file has one username that has read access and it is a root path /Access.

**Comment by adg** [comment]
@kushtrimjunuzi you should be able to delete the Access file as the tree user. Can you?

**Comment by kushtrimjunuzi** [comment]
@adg I tried delete from root user and from tree user (user+snapshot@domain.tld) and both successfully can delete files but not adding new files. Things that I can do now are, listing and dir creations.

**Comment by kushtrimjunuzi** [comment]
yes as I mention in prev comments, I can delete files and dirs without a problem from root user and snapshot.

**Comment by adg** [comment]
So *after* you delete the Access file, can you re-create it? If not, what are the error messages?

**Comment by kushtrimjunuzi** [comment]
no I'm not able to PUT files anymore, here is the error msg:

upspin: put: client.Put: permission denied:
store/remote("upspin.domain.tld:443").Put:
store/perm.Put: user upspin@domain.tld: user not authorized

as you can see my root user is same as domain where upspin is hosted, fyi

**Comment by kushtrimjunuzi** [comment]
I was able to fix the problem by removing upspin/server and re-setup upspin server. Trying to reproduce the issue and I couldn't, so not sure what happen but now is ok, anyway thanks for your time @adg
